Loop/Public/FeedbackLoop.swift

```diff
@@ -118,11 +118,51 @@ extension Loop {
             compacting transform: @escaping (SignalProducer<State, Never>) -> SignalProducer<U, Never>,
             effects: @escaping (U) -> Effect
         ) where Effect.Value == Event, Effect.Error == Never {
-            self.events = { state, output in
+            events = { state, output in
                 // NOTE: `observe(on:)` should be applied on the inner producers, so
                 //       that cancellation due to state changes would be able to
                 //       cancel outstanding events that have already been scheduled.
-                transform(state)
+                transform(state.map(\.0))
+                    .flatMap(.latest) { effects($0).producer.enqueue(to: output) }
+                    .start()
+            }
+        }
+
+        /// Creates a Feedback which re-evaluates the given effect every time the
+        /// `Signal` derived from the latest state yields a new value.
+        ///
+        /// If the previous effect is still alive when a new one is about to start,
+        /// the previous one would automatically be cancelled.
+        ///
+        /// - parameters:
+        ///   - transform: The transform which derives a `Signal` of values from the
+        ///                latest state.
+        ///   - effects: The side effect accepting transformed values produced by
+        ///              `transform` and yielding events that eventually affect
+        ///              the state.
+        public static func compacting<U, Effect: SignalProducerConvertible>(
```

Comment:
perhaps we should deprecate the `init` equivalents in favor of static factory API?